### ISSUE TITLE
[REQUEST] Muon → Gefen-Muon

### ISSUE THREAD

**Issue by ndvbd** [deepspeedai/DeepSpeed#8092]
Please consider adding support for Gefen-Muon. Same performance as Muon, with x4 less persistent optimizer memory footprint, and around 58% less peak optimizer memory footprint than Muon (when using the Muon version of PyTorch).

Reference implementation can be found here: https://github.com/ndvbd/Gefen:

pip install gefen
from gefen import GefenMuon

Same learning rate, and everything - one line of code change.

All experiments show exact same behaviour as Muon, but with decreased memory footprint, which enables larger batch size  and higher throughput because of that.

<img width="859" height="433" alt="Image" src="https://github.com/user-attachments/assets/134cf973-c184-4a3d-b59a-f117697aefdb" />

**Comment by the-DD5-is-too-damn-high** [comment]
Maintainers: recommend you hold off on this, the repo claims are not validated: 

[Issue #1 ](https://github.com/ndvbd/Gefen/issues/1#issuecomment-4802448972) (now deleted) highlighted gaps vs claims:

- broken FSDP (claimed working in repo/paper)
- missing paths for muon 1D matrix fallback (despite being touted as a "drop-in" replacement) 
- inability to reproduce VRAM savings and throughput increase claims
- learning rate parity claims  vs. adamw do not check out on modern models (e.g. qwen 3) 

FWIW - the repo authors seem to be banning people and deleting issues that counter their claims and manually pushing changes  instead of merging PRs with proper author attribution as well.   

[gefen.log](https://github.com/user-attachments/files/29399939/gefen.log)

[adamw.log](https://github.com/user-attachments/files/29399940/adamw.log)

**Comment by ndvbd** [comment]
Thank you for the respectful discussion. Feel free to contact me directly for any support, collaboration, or ideas. More engineering features and support for various distributed training setups and other specific configurations are definitely welcome, and will be added.
The current Gefen code is primarily targeted at model pre-training. So far, we have received case studies up to 32B parameters, with some reporting 6x shorter training time due to reduced CPU offloading.

As for Gefen-Muon, the Muon replacement, as it does not require second moment compression, so we also tested it on a fine-tuning task. Using Axolotl with the Qwen3 1.7B architecture on a single GPU, fine-tuned on Alpaca, we see almost identical training loss curves. Although more Muon to Gefen-Muon comparisons should be made, this suggests that Muon fine-tuning can potentially be replaced with Gefen-Muon. We will definitely update the repo when more case studies will come.

The optimizer peak memory consumption of Gefen-Muon was 27% less than Muon (3.7GiB vs 5.1Gib), and training curves looks almost the same. Note that Muon is applied on certain tensors (non embedding tensors, etc).

<img width="929" height="436" alt="Image" src="https://github.com/user-attachments/assets/5bd3983d-f142-4d3c-89cb-b649706f215b" />
